testing/test_transforms.py

```diff
@@ -0,0 +1,248 @@
+
+    @pytest.mark.parametrize('amin, amax, calId',
+                             [
+                                 (-100., 100., 0),
+                                 (0., 10., 1),
+                                 (0., 1e-10, 200)
+                             ])
+    def testCopy(self, amin, amax, calId):
+        tf = AccelTransform(amin=amin, amax=amax, calId=calId)
+
+        assert tf == tf.copy()
+
+
+class TestUnivariate:
+
+    def testInplace(self, univariate):
+        values = np.arange(5)
+        out = np.zeros(5)
+        univariate.inplace(values, out=out)
+
+        if len(univariate.coefficients) == 1:
+            a = univariate.coefficients[0]
+            np.testing.assert_array_equal(out, np.ones(5)*a)
+        elif len(univariate.coefficients) == 2:
+            a, b = univariate.coefficients
+            ref = univariate.references[0]
+            np.testing.assert_array_equal(out, a*(values - ref) + b)
+        else:
+            raise
+
+
+class TestBivariate:
+
+    @pytest.mark.parametrize('useTimestamp, noBivariates, useMean',
+                             [
+                                 (True,  True,  True),
+                                 (True,  True,  False),
+                                 (True,  False, True),
+                                 (True,  False, False),
+                                 (False, True,  True),
+                                 (False, True,  False),
+                                 (False, False, True),
+                                 (False, False, False),
+                                 ])
+    def testInplace(self, useTimestamp, noBivariates, useMean, bivariate, ssx66115):
+        """
+            Test inplace math for bivariate polynomials.  This one's more complex.
+
+            Parameters are for all combinations of using timestamps, using the
+            mean, and using bivariate calibration.
+
+            The formula is z = a*(x - xref)*(y - yref) + b*(x - xref) + c*(y - yref) + d
+
+            If noBivariates is `True`, then y must be 0.
+
+            If useMean is `True` or useTimestamp is `False`, y is the mean of
+            the secondary channel.
+
+            Otherwise, y is the value of the other channel at the timestamps.
+
+        """
+        eventList = ssx66115.channels[bivariate.channelId][bivariate.subchannelId].getSession()
+        if useTimestamp:
+            timestamps = np.linspace(
+                    ssx66115.channels[8].getSession()._blockTimes[0],
+                    ssx66115.channels[8].getSession()._blockTimes[-1],
+                    5,
+                    dtype=np.uint64,
+                    )
+            y = np.fromiter(
+                    (eventList[eventList.getEventIndexNear(t)][1] for t in timestamps),
+                    dtype=np.float64,
+                    )
+        else:
+            timestamps = None
+            y = eventList.getMean()
```

Comment:
so `useTimestamp=False, useMean=False` sets `y = eventList.getMean()`? From my alien eyes I'd assume `useMean=False` would make it so `getMean` doesn't get called; can you just confirm for me that this is intended?

Reply:
I also forgot to mention that I swapped this here for `y = 0` and the tests still passed. so maybe this isn't super critical?

Reply:
I've interpreted `useMean` as "use the mean instead of the timestamps", so `useMean=False` is not the same as `noBivariates=True`.

Reply:
as long as it's intentional that's fine